linkerd/proxy/transport/src/orig_dst.rs

```diff
@@ -117,23 +117,51 @@ mod linux {
     use tracing::warn;
 
     pub unsafe fn so_original_dst(fd: RawFd) -> io::Result<SocketAddr> {
+        let mut sockdomain: i32 = 0;
+        let mut sockdomain_len: libc::socklen_t = 32;
+
         let mut sockaddr: libc::sockaddr_storage = mem::zeroed();
-        let mut socklen: libc::socklen_t = mem::size_of::<libc::sockaddr_storage>() as u32;
+        let mut sockaddr_len: libc::socklen_t = mem::size_of::<libc::sockaddr_storage>() as u32;
+
+        let ret = libc::getsockopt(
+            fd,
+            libc::SOL_SOCKET,
+            libc::SO_DOMAIN,
+            &mut sockdomain as *mut _ as *mut _,
+            &mut sockdomain_len as *mut _ as *mut _,
+        );
+        if ret != 0 {
+            let e = io::Error::last_os_error();
+            warn!("failed to read SO_DOMAIN: {:?}", e);
+            return Err(e);
```

Comment:
I know that you're just being consistent with what is already in this function, but if we emit warnings here, these messages are going to be double-logged (since we are also returning an error). This will end up blowing up the accept() call and it will be logged as a server error.

We should probably remove the existing warning on line 160, too